### PR TITLE
more small const improvements

### PR DESCRIPTION
**Pull request by llogiq**

### CODE REVIEW

File: src/consts.rs [hunk 1]
```diff
@@ -149,15 +149,15 @@ impl PartialOrd for Constant {
...
-        LitBool(b) => ConstantBool(b),
+    match lit {
+        &LitStr(ref is, style) => ConstantStr(is.to_string(), style),
+        &LitBinary(ref blob) => ConstantBinary(blob.clone()),
+        &LitByte(b) => ConstantByte(b),
+        &LitChar(c) => ConstantChar(c),
+        &LitInt(value, ty) => ConstantInt(value, ty),
+        &LitFloat(ref is, ty) => ConstantFloat(is.to_string(), ty.into()),
+        &LitFloatUnsuffixed(ref is) => ConstantFloat(is.to_string(), FwAny),
+        &LitBool(b) => ConstantBool(b),
```
Comment (birkenfeld) [inline src/consts.rs]: Hm, these two hunks are reverts of changes in my previous PR. Intended?

File: src/consts.rs [hunk 2]
```diff
@@ -300,8 +300,8 @@ impl<'c, 'cc> ConstEvalContext<'c, 'cc> {
             ExprIf(ref cond, ref then, ref otherwise) =>
                 self.ifthenelse(&*cond, &*then, &*otherwise),
             ExprLit(ref lit) => Some(lit_to_constant(&lit.node)),
-            ExprVec(ref vec) => self.vec(&vec),
-            ExprTup(ref tup) => self.tup(&tup),
+            ExprVec(ref vec) => self.multi(&vec[..]).map(ConstantVec),
+            ExprTup(ref tup) => self.multi(&tup[..]).map(ConstantTuple),
```
Comment (birkenfeld) [inline src/consts.rs]: You can leave out the `[..]`, another change from my PR.

**Comment by llogiq** [comment]
Those chunks are testament to the fact that I don't know how to work with meld.

**Comment by Manishearth** [comment]
My suggestion for meld:

```
[merge]
tool = mymeld
conflictstyle = diff3
[mergetool "mymeld"]
cmd = meld $LOCAL $BASE $REMOTE --output $MERGED
```
in your gitconfig.

This will give you three windows, the center one being the common base, and left and right being local/remote with conflicts. Scroll up and down till you find conflicts, fix those. Conflicts come in the form of red lines or blue diff lines which come too close (i.e. line 70 is changed by remote, line 71 changed by base)

Once you finish fixing conflicts, click the center pane, `changes>merge all`, click the left pane, `changes>merge all from right`, click the right pane, `changes>merge all from left`, save the center pane (no need to save the `foo.rs.REMOTE` files, etc), and exit.

**Comment by llogiq** [comment]
Thanks, I'll try that.

**Comment by birkenfeld** [comment]
You *could* squash the two changes now, but otherwise r+

**Pull request merged**